### PR TITLE
ui: updated helm charts for hubble-ui 0.7

### PR DESCRIPTION
**Pull request by geakstr**
Signed-off-by: Dmitry Kharitonov <dmitry@isovalent.com>

### CODE REVIEW

**Comment by joestringer** [comment]
I think you need to run `make -C install/kubernetes` to generate the quick install / experimental install YAMLs based on the changes.

**Review by christarazi — COMMENTED** [review]
LGTM, please squash the two commit together, and it should be good to go.

File: install/kubernetes/cilium/charts/hubble-ui/values.yaml [hunk 1]
```diff
@@ -18,18 +26,17 @@ replicas: 1
...
+proxyResources: {}
 
 serviceAccount:
   annotations: {}
 
 ingress:
   enabled: false
-  annotations: {}
+  annotations:
+    {}
```
Comment (sayboras) [inline install/kubernetes/cilium/charts/hubble-ui/values.yaml]: nit the style here a little bit different with other places

File: install/kubernetes/cilium/charts/hubble-ui/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+---
+apiVersion: v1
```
Comment (sayboras) [inline install/kubernetes/cilium/charts/hubble-ui/templates/configmap.yaml:2]: dumb question, if this configmap changes, will envoy pick it up (e.g. similar to coredns) ?

File: install/kubernetes/cilium/charts/hubble-ui/templates/svc.yaml [hunk 3]
```diff
@@ -1,13 +1,13 @@
 kind: Service
 apiVersion: v1
 metadata:
-  namespace: {{ .Release.Namespace }}
+  namespace: {{.Release.Namespace}}
   name: hubble-ui
 spec:
   selector:
-    k8s-app: hubble-ui
+    app: hubble-ui
```
Comment (genbit) [inline install/kubernetes/cilium/charts/hubble-ui/templates/svc.yaml]: @geakstr needs to be 
```
k8s-app: hubble-ui 
```
in order for it to select relevant deployment

**Review by genbit — CHANGES_REQUESTED** [review]

File: install/kubernetes/cilium/charts/hubble-ui/templates/svc.yaml [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
   ports:
     - name: http
       port: 80
-      targetPort: 12000
+      targetPort: 8081
```
Comment (sayboras) [inline install/kubernetes/cilium/charts/hubble-ui/templates/svc.yaml:12]: we need to update few docs for this port changes as well ? For example, https://github.com/cilium/cilium/blob/master/Documentation/gettingstarted/hubble-enable.rst

**Review by sayboras — COMMENTED** [review]
Few minor points

File: install/kubernetes/cilium/charts/hubble-ui/templates/svc.yaml (hunk 4 above)
Comment (genbit) [reply install/kubernetes/cilium/charts/hubble-ui/templates/svc.yaml:12]: no, port-forward command will continue to work

File: install/kubernetes/experimental-install.yaml [hunk 5]
```diff
@@ -800,30 +867,54 @@ spec:
...
               name: http
           resources:
             {}
+          command: ["envoy"]
+          args:
+            [
+              "-c",
+              "/etc/envoy.yaml",
+              "-l",
+              "debug"
```
Comment (genbit) [inline install/kubernetes/experimental-install.yaml]: probably need to lower the logs level

**Review by genbit — CHANGES_REQUESTED** [review]

File: install/kubernetes/cilium/charts/hubble-ui/templates/deployment.yaml [hunk 6]
```diff
@@ -1,10 +1,10 @@
 kind: Deployment
 apiVersion: apps/v1
 metadata:
-  namespace: {{ .Release.Namespace }}
+  namespace: {{.Release.Namespace}}
```
Comment (fristonio) [inline install/kubernetes/cilium/charts/hubble-ui/templates/deployment.yaml]: I think this change is not required.

File: install/kubernetes/cilium/charts/hubble-ui/templates/deployment.yaml [hunk 7]
```diff
@@ -13,29 +13,50 @@ spec:
...
+        - name: frontend
+          image: "{{ .Values.frontendImage.repository }}:{{ .Values.frontendImage.tag }}"
+          imagePullPolicy: {{.Values.frontendImage.pullPolicy}}
           ports:
-            - containerPort: 12000
+            - containerPort: 8080
               name: http
           resources:
-            {{- toYaml .Values.resources | trim | nindent 12 }}
+            {{- toYaml .Values.frontendResources | trim | nindent 12}}
```
Comment (fristonio) [inline install/kubernetes/cilium/charts/hubble-ui/templates/deployment.yaml]:
```suggestion
            {{- toYaml .Values.frontendResources | trim | nindent 12 }}
```

File: install/kubernetes/cilium/charts/hubble-ui/templates/deployment.yaml [hunk 8]
```diff
@@ -1,10 +1,10 @@
 kind: Deployment
 apiVersion: apps/v1
 metadata:
-  namespace: {{ .Release.Namespace }}
+  namespace: {{.Release.Namespace}}
   name: hubble-ui
 spec:
-  replicas: {{ .Values.replicas }}
+  replicas: {{.Values.replicas}}
```
Comment (fristonio) [inline install/kubernetes/cilium/charts/hubble-ui/templates/deployment.yaml]: Here as well.

File: install/kubernetes/cilium/charts/hubble-ui/templates/deployment.yaml [hunk 9]
```diff
@@ -13,29 +13,50 @@ spec:
...
           ports:
-            - containerPort: 12000
+            - containerPort: 8080
               name: http
           resources:
-            {{- toYaml .Values.resources | trim | nindent 12 }}
+            {{- toYaml .Values.frontendResources | trim | nindent 12}}
+        - name: backend
+          image: "{{ .Values.backendImage.repository }}:{{ .Values.backendImage.tag }}"
+          imagePullPolicy: {{.Values.backendImage.pullPolicy}}
```
Comment (fristonio) [inline install/kubernetes/cilium/charts/hubble-ui/templates/deployment.yaml]:
```suggestion
          imagePullPolicy: {{ .Values.backendImage.pullPolicy }}
```
I think we should keep the formatting consistent.

**Review by fristonio — CHANGES_REQUESTED** [review]
Mostly looks good :rocket: Just some minor nits around formatting.

File: install/kubernetes/cilium/charts/hubble-ui/templates/deployment.yaml (hunk 6 above)
Comment (fristonio) [reply install/kubernetes/cilium/charts/hubble-ui/templates/deployment.yaml]: I think we have a few more formatting issues that I forgot to highlight here. :sweat_smile:

File: install/kubernetes/cilium/charts/hubble-ui/values.yaml [hunk 10]
```diff
@@ -1,12 +1,20 @@
...
+  pullPolicy: IfNotPresent
+backendImage:
+  repository: quay.io/cilium/hubble-ui-backend
+  # tag is the container image tag to use.
+  # Ref: https://github.com/cilium/hubble-ui/releases
+  # Ref: https://quay.io/repository/cilium/hubble-ui-backend?tab=tags
+  tag: v0.7.1
+  pullPolicy: IfNotPresent
+proxyImage:
+  repository: envoyproxy/envoy
```
Comment (aanm) [inline install/kubernetes/cilium/charts/hubble-ui/values.yaml]: `docker.io/envoyproxy/envoy`

**Review by aanm — CHANGES_REQUESTED** [review]
We can also have a dedicated file for the envoy configuration and import it in the configmap.
```patch
diff --git a/install/kubernetes/cilium/charts/hubble-ui/files/envoy/envoy.yaml b/install/kubernetes/cilium/charts/hubble-ui/files/envoy/envoy.yaml
new file mode 100644
index 000000000..31c778b61
--- /dev/null
+++ b/install/kubernetes/cilium/charts/hubble-ui/files/envoy/envoy.yaml
@@ -0,0 +1,58 @@
+static_resources:
+  listeners:
+    - name: listener_hubble_ui
+      address:
+        socket_address:
+          address: 0.0.0.0
+          port_value: 8081
+      filter_chains:
+        - filters:
+            - name: envoy.filters.network.http_connection_manager
+              config:
+                codec_type: auto
+                stat_prefix: ingress_http
+                route_config:
+                  name: local_route
+                  virtual_hosts:
+                    - name: local_service
+                      domains: ['*']
+                      routes:
+                        - match:
+                            prefix: '/api/'
+                          route:
+                            cluster: backend
+                            max_grpc_timeout: 0s
+                            prefix_rewrite: '/'
+                        - match:
+                            prefix: '/'
+                          route:
+                            cluster: frontend
+                      cors:
+                        allow_origin_string_match:
+                          - prefix: '*'
+                        allow_methods: GET, PUT, DELETE, POST, OPTIONS
+                        allow_headers: keep-alive,user-agent,cache-control,content-type,content-transfer-encoding,x-accept-content-transfer-encoding,x-accept-response-streaming,x-user-agent,x-grpc-web,grpc-timeout
+                        max_age: '1728000'
+                        expose_headers: grpc-status,grpc-message
+                http_filters:
+                  - name: envoy.filters.http.grpc_web
+                  - name: envoy.filters.http.cors
+                  - name: envoy.filters.http.router
+  clusters:
+    - name: frontend
+      connect_timeout: 0.25s
+      type: strict_dns
+      lb_policy: round_robin
+      hosts:
+        - socket_address:
+            address: 127.0.0.1
+            port_value: 8080
+    - name: backend
+      connect_timeout: 0.25s
+      type: logical_dns
+      lb_policy: round_robin
+      http2_protocol_options: {}
+      hosts:
+        - socket_address:
+            address: 127.0.0.1
+            port_value: 8090
diff --git a/install/kubernetes/cilium/charts/hubble-ui/templates/configmap.yaml b/install/kubernetes/cilium/charts/hubble-ui/templates/configmap.yaml
index bc43ee70c..e69517e6b 100644
--- a/install/kubernetes/cilium/charts/hubble-ui/templates/configmap.yaml
+++ b/install/kubernetes/cilium/charts/hubble-ui/templates/configmap.yaml
@@ -5,62 +5,4 @@ metadata:
   name: hubble-ui-envoy
   namespace: {{ .Release.Namespace }}
 data:
-  envoy.yaml: >
-    static_resources:
-      listeners:
-        - name: listener_hubble_ui
-          address:
-            socket_address:
-              address: 0.0.0.0
-              port_value: 8081
-          filter_chains:
-            - filters:
-                - name: envoy.filters.network.http_connection_manager
-                  config:
-                    codec_type: auto
-                    stat_prefix: ingress_http
-                    route_config:
-                      name: local_route
-                      virtual_hosts:
-                        - name: local_service
-                          domains: ['*']
-                          routes:
-                            - match:
-                                prefix: '/api/'
-                              route:
-                                cluster: backend
-                                max_grpc_timeout: 0s
-                                prefix_rewrite: '/'
-                            - match:
-                                prefix: '/'
-                              route:
-                                cluster: frontend
-                          cors:
-                            allow_origin_string_match:
-                              - prefix: '*'
-                            allow_methods: GET, PUT, DELETE, POST, OPTIONS
-                            allow_headers: keep-alive,user-agent,cache-control,content-type,content-transfer-encoding,x-accept-content-transfer-encoding,x-accept-response-streaming,x-user-agent,x-grpc-web,grpc-timeout
-                            max_age: '1728000'
-                            expose_headers: grpc-status,grpc-message
-                    http_filters:
-                      - name: envoy.filters.http.grpc_web
-                      - name: envoy.filters.http.cors
-                      - name: envoy.filters.http.router
-      clusters:
-        - name: frontend
-          connect_timeout: 0.25s
-          type: strict_dns
-          lb_policy: round_robin
-          hosts:
-            - socket_address:
-                address: 127.0.0.1
-                port_value: 8080
-        - name: backend
-          connect_timeout: 0.25s
-          type: logical_dns
-          lb_policy: round_robin
-          http2_protocol_options: {}
-          hosts:
-            - socket_address:
-                address: 127.0.0.1
-                port_value: 8090
+{{ (.Files.Glob "files/envoy/*").AsConfig | indent 2 }}
\ No newline at end of file
diff --git a/install/kubernetes/experimental-install.yaml b/install/kubernetes/experimental-install.yaml
index ce90557df..af1701c93 100644
--- a/install/kubernetes/experimental-install.yaml
+++ b/install/kubernetes/experimental-install.yaml
@@ -184,7 +184,7 @@ metadata:
   name: hubble-ui-envoy
   namespace: kube-system
 data:
-  envoy.yaml: >
+  envoy.yaml: |
     static_resources:
       listeners:
         - name: listener_hubble_ui
```

**Comment by geakstr** [comment]
Cool, thanks

**Review by aanm — APPROVED** [review]

**Review by fristonio — APPROVED** [review]
LGTM

**Comment by aanm** [comment]
hit https://github.com/cilium/cilium/issues/13198

**Pull request merged**